### PR TITLE
feat(TKC-1642): add TestWorkflow support of cloning Git, executing tests, and artifacts

### PR DESCRIPTION
**Pull request by rangoo94**
## Pull request description 

* Add `testkube-toolkit` image, that supports `clone`, `execute`, and `artifacts` commands
* Add corresponding `content.git`, `execute`, and `artifacts` support in the TestWorkflow
* Add API and CLI for listing and fetching artifacts
* Minor fixes for TestWorkflows

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Fixes

- https://linear.app/kubeshop/issue/TKC-1642/testworkflows-build-and-use-orchestration-toolkit

### CODE REVIEW

File: cmd/tcl/testworkflow-toolkit/artifacts/cloud_uploader.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	}
+	for _, r := range d.reqEnhancers {
+		r(req, path, size)
+	}
+	req.ContentLength = size
+	if req.Header.Get("Content-Type") == "" {
+		req.Header.Set("Content-Type", "application/octet-stream")
+	}
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```
[the remark this answers is not in the document]
Comment (rangoo94) [reply cmd/tcl/testworkflow-toolkit/artifacts/cloud_uploader.go:93]: Sure thing, just at the moment I don't have any idea of even moderately simple way of providing certificate bundles as volumes

File: cmd/tcl/testworkflow-toolkit/artifacts/tar_stream.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"sync"
+)
+
+type tarStream struct {
+	r  io.ReadCloser
```
Comment (exu) [inline cmd/tcl/testworkflow-toolkit/artifacts/tar_stream.go:21]: can we name it somehow - quite unreadable
e.g. tarCachedProcessor have usual names

File: cmd/tcl/testworkflow-toolkit/commands/clone.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+		Args:  cobra.ExactArgs(2),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			uri, err := url.Parse(args[0])
+			ui.ExitOnError("repository uri", err)
+			outputPath, err := filepath.Abs(args[1])
+			ui.ExitOnError("output path", err)
+
+			// Disable interactivity
+			os.Setenv("GIT_TERMINAL_PROMPT", "0")
```
Comment (exu) [inline cmd/tcl/testworkflow-toolkit/commands/clone.go:43]: would be better to use local cmd.Env than setting it globally.

Comment (rangoo94) [reply cmd/tcl/testworkflow-toolkit/commands/clone.go:43]: To be honest, I'm not sure if the current solution isn't better though - putting it anywhere else will masquerade it, which may be less obvious during maintenance

File: pkg/tcl/apitcl/v1/testworkflowexecutions.go [hunk 4]
```diff
@@ -281,6 +282,82 @@ func (s *apiTCL) AbortAllTestWorkflowExecutionsHandler() fiber.Handler {
...
+	}
+}
+
+func (s *apiTCL) GetTestWorkflowArtifactHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		executionID := c.Params("executionID")
+		fileName := c.Params("filename")
+		errPrefix := fmt.Sprintf("failed to get artifact %s for workflow execution %s", fileName, executionID)
+
+		// TODO fix this someday :) we don't know 15 mins before release why it's working this way
```
Comment (exu) [inline pkg/tcl/apitcl/v1/testworkflowexecutions.go:310]: :D

**Review by exu — APPROVED** [review]
LGTM - some nitpicks

File: cmd/tcl/testworkflow-toolkit/artifacts/cloud_uploader.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rangoo94) [reply cmd/tcl/testworkflow-toolkit/artifacts/cloud_uploader.go:93]: @vsukhin - I remember that we have lately added support for that in existing code, but I cannot find it:
* There are `STORAGE_CERT/KEY/CA_FILE` variables
* These variables are used only in API Server in the Logs Service and API Server
* I cannot find any way to provide it to the existing Artifacts Scraper - is it possible some way?

[the remark this answers is not in the document]
Comment (rangoo94) [reply cmd/tcl/testworkflow-toolkit/artifacts/cloud_uploader.go:93]: Generally I would see it the way:

> If there is a configured certificate bundle:
> * Fetch these files into API Server memory
> * Push them to some ConfigMap
> * Mount this ConfigMap to the volumes, and point their env variables into these files

**Pull request merged**